activesupport/lib/active_support/core_ext/module/delegation.rb

```diff
@@ -203,34 +209,33 @@ def delegate(*methods, to: nil, prefix: nil, allow_nil: nil, private: nil)
       # whereas conceptually, from the user point of view, the delegator should
       # be doing one call.
       if allow_nil
-        method_def = [
-          "def #{method_prefix}#{method}(#{definition})",
-          "  _ = #{to}",
-          "  if !_.nil? || nil.respond_to?(:#{method})",
-          "    _.#{method}(#{definition})",
-          "  end",
+        method = method.to_s
+
+        method_def <<
+          "def #{method_name}(#{definition})" <<
```

Comment:
Can we keep the tailing spaces? They make the method easily alienable

Reply:
What about squiggly heredocs? Can we use them here?

Reply:
@rafaelfranca I don't understand what you mean by "alienable", easy to align?

I removed them so that we keep less different string, e.g. `"  end"` and `"end"` both become `"end"`. It's a very small optimization, so I can totally remove it if you feel it makes the code nicer.

@kaspth yes and no. `Module#delegate` want to to define all the methods on a single line (`def foo; blah; end`) so that in backtraces everything will be attributed to the original `delegate :foo, to: :bar` statement.

So we can use heredocs, but then we need to apply `.tr("\n", ";")`, which I benchmarked to be somewhat slower than the current join. Not by that much though so again, I'm open to it.

Reply:
Yes, easier to align. If we want to keep that optimization I'd just remove the artificial alignment that you added and put every string in the same column.

Right now if you run a command to realign the file it will change those lines and we would need to keep aligning those lines manually.

Reply:
Done.